src/Caller.sol

```diff
@@ -101,36 +110,47 @@ contract Caller is EIP712("Caller", "1"), ERC2771Context(address(this)) {
     /// @param unauthorized The authorized address.
     event Unauthorized(address indexed sender, address indexed unauthorized);
 
+    /// @notice Emitted when revoking all authorizations to make calls on behalf of the `sender`.
+    /// @param sender The authorizing address.
+    event UnauthorizedAll(address indexed sender);
+
     /// @notice Grants the authorization of an address to make calls on behalf of the sender.
     /// @param user The authorized address.
     function authorize(address user) public {
         address sender = _msgSender();
-        require(_authorized[sender].add(user), "Address already is authorized");
+        require(_getAuthorizedSet(sender).add(user), "Address already is authorized");
         emit Authorized(sender, user);
     }
 
     /// @notice Revokes the authorization of an address to make calls on behalf of the sender.
     /// @param user The unauthorized address.
     function unauthorize(address user) public {
         address sender = _msgSender();
-        require(_authorized[sender].remove(user), "Address is not authorized");
+        require(_getAuthorizedSet(sender).remove(user), "Address is not authorized");
         emit Unauthorized(sender, user);
     }
 
+    /// @notice Revokes all authorizations to make calls on behalf of the sender.
+    function unauthorizeAll() public {
+        address sender = _msgSender();
```

Comment:
```suggestion
        address sender = msg.sender;
        require(msg.sender != address(this));
```
What would we lose if we restrict it to the actual sender?

Reply:
We'd lose the flexibility, like being able to batch or call via an operator. It wouldn't change much with the security model, because it's still possible to call `unauthorize` in a loop, it's just more expensive.

Reply:
Yeah, for the `unauthorize` this make sense but for  `unauthorizeAll` ?

Reply:
It's the same for both of them, they are perfectly equivalent to each other. If you restrict either one of them, the other can be used as a less gas efficient substitute.

Reply:
Yeah, fair point. Let's keep it :+1: